dask_image/ndinterp/__init__.py

```diff
@@ -0,0 +1,236 @@
+
+from scipy.ndimage import affine_transform as ndimage_affine_transform
+import warnings
+
+
+from ..dispatch._dispatch_ndinterp import (
+    dispatch_affine_transform,
+    dispatch_asarray,
+)
+
+
+__all__ = [
+    "affine_transform",
+]
+
+
+def affine_transform(
+        image,
+        matrix,
+        offset=None,
+        output_shape=None,
+        order=1,
+        output_chunks=None,
+        **kwargs
+):
+    """Apply an affine transform using Dask. For every
+    output chunk, only the slice containing the relevant part
+    of the image is processed. Chunkwise processing is performed
+    either using `ndimage.affine_transform` or
+    `cupyx.scipy.ndimage.affine_transform`, depending on the input type.
+
+    Notes
+    -----
+        Differences to `ndimage.affine_transformation`:
+        - currently, prefiltering is not supported
+          (affecting the output in case of interpolation `order > 1`)
+        - default order is 1
+        - modes 'reflect', 'mirror' and 'wrap' are not supported
+
+        Arguments equal to `ndimage.affine_transformation`,
+        except for `output_chunks`.
+
+    Parameters
+    ----------
+    image : array_like (Numpy Array, Cupy Array, Dask Array...)
+        The image array.
+    matrix : array (ndim,), (ndim, ndim), (ndim, ndim+1) or (ndim+1, ndim+1)
+        Transformation matrix.
+    offset : array (ndim,)
+        Transformation offset.
+    output_shape : array (ndim,), optional
+        The size of the array to be returned.
+    order : int, optional
+        The order of the spline interpolation. Note that for order>1
+        scipy's affine_transform applies prefiltering, which is not
+        yet supported and skipped in this implementation.
+    output_chunks : array (ndim,), optional
+        The chunks of the output Dask Array.
+
+    Returns
+    -------
+    affine_transform : Dask Array
+        A dask array representing the transformed output
+
+    """
+
+    if not type(image) == da.core.Array:
+        image = da.from_array(image)
+
+    if output_shape is None:
+        output_shape = image.shape
+
+    if output_chunks is None:
+        output_chunks = image.shape
+
+    # Perform test run to ensure parameter validity.
+    ndimage_affine_transform(np.zeros([0] * image.ndim),
+                             matrix,
+                             offset)
+
+    # Make sure parameters contained in matrix and offset
+    # are not overlapping, i.e. that the offset is valid as
+    # it needs to be modified for each chunk.
+    # Further parameter checks are performed directly by
+    # `ndimage.affine_transform`.
+
+    matrix = np.asarray(matrix)
+    offset = np.asarray(offset).squeeze()
+
+    # these lines were copied and adapted from `ndimage.affine_transform`
+    if (matrix.ndim == 2 and matrix.shape[1] == image.ndim + 1 and
+            (matrix.shape[0] in [image.ndim, image.ndim + 1])):
+        # if matrix.shape[0] == image.ndim + 1:
```

Comment:
I notice lines 100-107 are commented out, do these need to be removed?

Reply:
Yes thanks, I had accidentally left them there. That warning is instead raised by the `ndimage` test run in line 84:

```python
    ndimage_affine_transform(np.zeros([0] * image.ndim),
                             matrix,
                             offset)
```